core/field_variable.js

```diff
@@ -75,7 +75,7 @@ Blockly.FieldVariable.prototype.initModel = function() {
         this.sourceBlock_.isInFlyout ?
             this.sourceBlock_.workspace.targetWorkspace :
             this.sourceBlock_.workspace;
-    this.setValue(Blockly.Variables.generateUniqueName(workspace));
+    this.setText(Blockly.Variables.generateUniqueName(workspace));
```

Comment:
SetText doesn't update the value of the block (setValue updates the text) so I think you'll decouple text and id here by accident.